src/Form/RedirectForm.php

```diff
@@ -133,7 +117,7 @@ public function validate(array $form, FormStateInterface $form_state) {
     $parsed_url = UrlHelper::parse(trim($source['path']));
     $path = isset($parsed_url['path']) ? $parsed_url['path'] : NULL;
     $query = isset($parsed_url['query']) ? $parsed_url['query'] : NULL;
-    $hash = Redirect::generateHash($path, $query, $form_state->getValue('language'));
+    $hash = Redirect::generateHash($path, $query, $form_state->getValue('language')[0]['value']);
```

Comment:
I couldn't get the object returned from `parent::validate()` to work. The item list inside of `redirect_source` was always an empty array (`redirect_redirect` had proper data).

I've added the separate test case that simply runs with the additional 2 modules enabled.